icu4j/main/core/src/main/java/com/ibm/icu/impl/EraRules.java

```diff
@@ -116,68 +126,103 @@ public static EraRules getInstance(CalType calType, boolean includeTentativeEra)
             }
         }
 
+        // Now make array of just the eras we have rules for, with startDate and eraCode for each
+        int[][] startDateAndEras = new int[numEras][2];
+        int startDateIdx = 0;
+        for (int eraIdx = 0; eraIdx < eraStartDates.size(); eraIdx++) {
+            if (isSet(eraStartDates.get(eraIdx).intValue())) {
+                if (startDateIdx >= numEras) {
+                    throw new ICUException(
+                            "With start date for era code " + eraIdx + ", found more start dates than era rule bundles (" + numEras + ")");
+                }
+                startDateAndEras[startDateIdx][0] = eraStartDates.get(eraIdx).intValue();
+                startDateAndEras[startDateIdx][1] = eraIdx;
+                startDateIdx++;
+            }
+        }
+        if (startDateIdx != numEras) {
```

Comment:
at most assert, I think

Reply:
Probably another instance of overdoing the checks, I think this is actually not possible.

Reply:
Removed here and in C